### DISCUSSION TITLE
Opening neovim docs in the current window instead of a split

### DISCUSSION THREAD

**Discussion by aslamosama** [ibhagwan/fzf-lua#1268] [Q&A] [answered]
If I select an option from the `helptags` it opens in a horizontal split by default. Is there a way to make it open in full screen by default. I found a custom function `help-curwin` in neovim docs but I have no idea how to use it with fzf.lua `helptags`.

**Comment by ibhagwan** [comment]
That behavior is controlled by neovim, run `:help` and it will be the same behavior, you can open in a vertical split by running `:vert help` but I’m not sure what/if there’s a command to open directly in full screen mode. You can however create a custom fzf-lua action that opens the help and then executes `<C-w><C-o>` to make the new buffer the only window.

**Reply by aslamosama** [reply]
I found `help-curwin` in neovim docs. and adding this to my init.lua I was able to use `:HelpCurwin` to open help in fullscreen.
```
vim.cmd([[
command -bar -nargs=? -complete=help HelpCurwin call HelpCurwin(<q-args>)

let g:did_open_help = v:false

function! HelpCurwin(subject) abort
  let mods = 'silent noautocmd keepalt'
  if !g:did_open_help
    execute mods .. ' help'
    execute mods .. ' helpclose'
    let g:did_open_help = v:true
  endif
  if !empty(getcompletion(a:subject, 'help'))
    execute mods .. ' edit ' .. &helpfile
    set buftype=help
  endif
  return 'help ' .. a:subject
endfunction
]])
```

**Reply by ibhagwan** [reply, marked as answer]
That’s great, you can do the same with fzf-lua by copying the current action (2 lines) and changing the cmd to whatever you wanted and set the new action as your help default action:
https://github.com/ibhagwan/fzf-lua/blob/06d4fe9f9f86c40756b665b535b7d1dd6f25b639/lua/fzf-lua/actions.lua#L569-L572

**Reply by tssm** [reply]
Hey @ibhagwan,

would you be interested in a PR implementing this behind an option? I have [my own patch](https://github.com/tssm/.config/blob/master/nixpkgs/overlays/fzf-lua-help-in-current-window.patch) that I've been using for a while, so I would just need to wrap it.

([I made a similar proposal for Telescope](https://github.com/nvim-telescope/telescope.nvim/issues/2080), but I never got an answer)

**Reply by ibhagwan** [reply]
> Hey @ibhagwan,
> 
> would you be interested in a PR implementing this behind an option? I have [my own patch](https://github.com/tssm/.config/blob/master/nixpkgs/overlays/fzf-lua-help-in-current-window.patch) that I've been using for a while, so I would just need to wrap it.
> 
> ([I made a similar proposal for Telescope](https://github.com/nvim-telescope/telescope.nvim/issues/2080), but I never got an answer)

Sure why not, but  not as a new action users can configure instead of the default, I wouldn’t change the default behavior at this point.
